### PR TITLE
Adding travis.yml file For build integration 

### PR DESCRIPTION
**Pull request by sherrif10**
Kindly review  thanks
 cc @k-joseph  @dkayiwa

### CODE REVIEW

**Comment by sherrif10** [comment]
This will fail because the module doesn't run successful due to some constraint that is being worked up on, so ideally its good to merge it and continue resolving those failures cc @k-joseph  @dkayiwa

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: java
+
```
Comment (kaweesi) [inline .travis.yml:2]: We are at the moment integrating directly with bamboo ci and haven't included travis on our architecture

Comment (sherrif10) [reply .travis.yml:2]: ohh thanks for your feedback

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+language: java
```
Comment (kaweesi) [inline .travis.yml:1]: add the commands to run the tests

Comment (sherrif10) [reply .travis.yml:1]: @kaweesi  Have addressed and updated the pull request

**Comment by sherrif10** [comment]
Hello @kaweesi  any  more suggestion on this

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+jdk:
+ - openjdk8
+ 
+ cache:
+  directories:
+  - $HOME/.m2
+
+script:
+  - mvn clean install
+  - mvn clean test
```
Comment (kaweesi) [inline .travis.yml:12]: these 2 are running tests and so no need of the second

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+notifications:
+  email: false
+  irc:
+    channels:
+      - "irc.freenode.org#openmrs"
```
Comment (kaweesi) [inline .travis.yml:18]: we don't need to notify IRC at every PR

File: .travis.yml (hunk 3 above)
Comment (kaweesi) [reply .travis.yml:12]: can you lauch a docker ref app instance and point the tests properties to it in here

**Review by kaweesi — CHANGES_REQUESTED** [review]
thanks @sherrif10 for this, please checkout my reviews in comments, thanks

**Comment by sherrif10** [comment]
Thanks let me resolve them

File: .travis.yml (hunk 4 above)
Comment (sherrif10) [reply .travis.yml]: Resolved thanks

File: .travis.yml (hunk 3 above)
Comment (sherrif10) [reply .travis.yml]: Some more clarification about this, Are you meaning of testing this commit against running docker  refapp instance

Comment (sherrif10) [reply .travis.yml]: This may not work when trying to test it out , not sure if i answer your question thanks

Comment (kaweesi) [reply .travis.yml]: you need the first of those 2 commands. and yes, we need to lauch ref app within travis, wait fo rit and test against it

Comment (sherrif10) [reply .travis.yml]: Have only included only one command thanks

Comment (sherrif10) [reply .travis.yml]: Seems am mistakenly understanding your point, isnt this only testing the build only

**Comment by sherrif10** [comment]
cc @kaweesi  any more suggestions on this commits

**Comment by kaweesi** [comment]
BTW, could you first fix distro's master builds, currently the tests are skipped, start from fixing that so that the ui tests run in travis without issues in distro repo

**Comment by sherrif10** [comment]
> currently the tests are skipped

sure though i thought this ticket was to handle any build that is pushed via  any commit  in qa-contrib, 


> start from fixing that so that the ui tests run in travis without issues in distro repo

 sure thanks am handling this in on of the ticket am currently working on

**Pull request closed without merging**